### ISSUE TITLE
guarded resources are incorrectly included in the coverage report as untouched resources

### ISSUE THREAD

**Issue by y0y0z** [chefspec/chefspec#847]
i am using `chefspec` version 6.2.0 to test my recipes. when i run chefspec, i see that all of the tests passed and that i have 100% coverage. upgrading to higher versions (7.0.0, and 7.1.0), causing the touch coverage to drop.

here are more details.
```bash
$ cat my_cookbook/metadata.rb | grep -i sudo
depends 'sudo', '3.5.0'

$ cat my_cookbook/recipes/my_recipe.rb
include_recipe 'sudo'

users_manage 'my_group' do
  not_if { some_library_func_that_returns_true }
end

sudo 'my_group' do
  not_if { some_library_func_that_returns_true }
end

$ cat my_cookbook/spec/recipes/my_recipe_spec.rb
require 'chefspec'

describe 'my_cookbook::my_recipe' do
  cached(:chef_run) { chef_run_converge }

  it 'includes recipes' do
    expect(chef_run).to include_recipe('sudo')
  end

  it 'does not create my_group users as sudoers' do
    expect(chef_run).not_to create_users_manage('my_group')
    expect(chef_run).not_to install_sudo('my_group')
  end
end
```

the `some_library_func_that_returns_true` also returns `true` when chefspec runs.

when i run chefspec 6.2.0, all tests have passed and recipe resources have been touched. 
when i run chefspec higher than 6.2.0, all tests passed though recipe resources were not touched.

any idea what is wrong and what should be fixed?

**Comment by y0y0z** [comment]
trying to bump this thread to get more attention

**Comment by y0y0z** [comment]
@sethvargo i know you are extra busy, though i would really like to enjoy the latest chefspec version. could you please share your insight on the issue above?

TL;DR: some tests needed to be modified for chefspec > 6.2.0. although tests are passing, the touch coverage drops.

**Comment by y0y0z** [comment]
@coderanger trying to get your attention on this one. thank you in advanced.

**Comment by y0y0z** [comment]
anyone?

**Comment by lamont-granquist** [comment]
still a bug in 7.2.1

**Comment by lamont-granquist** [comment]
```
% cat recipes/default.rb

file "/etc/whatever.conf" do
  action :create
  not_if { true }
end
```

```
% cat spec/spec_helper.rb
# spec_helper.rb
require 'chefspec'

ChefSpec::Coverage.start!
```

```
% cat spec/test_spec.rb
require 'spec_helper'

describe 'test::default' do
  cached(:chef_run) { ChefSpec::SoloRunner.new { |node| }.converge(described_recipe) }
  subject { chef_run }

  it { expect(chef_run).not_to create_file("/etc/whatever.conf") }
end
```

**Comment by coderanger** [comment]
Coverage reporting is deprecated and will be removed, any bugs in it are not likely to get fixed.
